connectors/src/connectors/zendesk/temporal/workflows.ts

```diff
@@ -446,6 +447,12 @@ export async function zendeskGarbageCollectionWorkflow({
     } while (cursor !== null);
   }
 
+  // deleting the categories that have no permission anymore
+  hasMore = true;
+  while (hasMore) {
+    hasMore = await removeForbiddenCategoriesActivity(connectorId);
```

Comment:
A pattern we use it have one activity that fetches the next ids to handle, and then an activity that handles the ids returned by the first. If you want to do it as is with one activity this is fine but I would suggest having it return `{ hasMore: boolean }` instead of `boolean` just to clarify what the function return is used for.

Reply:
sounds good, I do this at a few other places, ok for you if I do this in another PR?

Reply:
Sure!